src/Orleans/IDs/ActivationId.cs

```diff
@@ -69,9 +69,17 @@ public static ActivationId GetSystemActivation(GrainId grain, SiloAddress locati
             return FindOrCreate(grain.Key);
         }
 
-        internal static ActivationId GetActivationId(GrainId grain)
+        public static ActivationId GetClientGWActivation(GrainId grain, SiloAddress location)
         {
-            return FindOrCreate(grain.Key);
+            if (!grain.IsClient)
+                throw new ArgumentException("ClientGW activation IDs can only be created for system grains");
```

Comment:
nit:  Not clear how error message "ClientGW activation IDs can only be created for system grains" relates to IsClient check?